### PR TITLE
Add payload bytes interface.

### PR DESCRIPTION
**Pull request by godchen0212**
Add payload bytes interface.
/kind enhancement
Issue #8937 

Signed-off-by: godchen0212 <qingxiang.chen@zilliz.com>

### CODE REVIEW

File: internal/storage/payload.go [hunk 1]
```diff
@@ -176,6 +178,18 @@ func (w *PayloadWriter) AddBoolToPayload(msgs []bool) error {
 	return HandleCStatus(&status, "AddBoolToPayload failed")
 }
 
+func (w *PayloadWriter) AddByteToPayload(msgs []byte) error {
+	length := len(msgs)
+	if length <= 0 {
+		return errors.New("can't add empty msgs into payload")
+	}
+	cMsgs := (*C.int8_t)(unsafe.Pointer(&msgs[0]))
```
Comment (DragonDriver) [inline internal/storage/payload.go:186]: better to be `uint8_t`.

Comment (DragonDriver) [reply internal/storage/payload.go:186]: For almost every programming languages, `byte` corespond to `uint8_t`.

File: internal/storage/payload.go [hunk 2]
```diff
@@ -176,6 +178,18 @@ func (w *PayloadWriter) AddBoolToPayload(msgs []bool) error {
...
+	cLength := C.int(length)
+
+	status := C.AddInt8ToPayload(w.payloadWriterPtr, cMsgs, cLength)
```
Comment (DragonDriver) [inline internal/storage/payload.go:189]: Better to have another `AddUint8ToPayload` interface. I noticed we use `arrow::Int8Builder` to write data. Not sure if there are some issues we also use `arrow::Int8Builder` to store `byte(uint8_t)`.

File: internal/storage/payload.go (hunk 1 above)
Comment (czs007) [reply internal/storage/payload.go:186]: Can we add a new data type  DataType_BLOB which means  []byte in go or  uint8 array in C?

And add AddUint8ToPayload methods.

@godchen0212 @DragonDriver

Comment (czs007) [reply internal/storage/payload.go:186]: Copied from mysql documentation.

In most respects, you can regard a BLOB column as a VARBINARY column that can be as large as you like. Similarly, you can regard a TEXT column as a VARCHAR column. BLOB and TEXT differ from VARBINARY and VARCHAR in the following ways:

1. For indexes on BLOB and TEXT columns, you must specify an index prefix length. For CHAR and VARCHAR, a prefix length is optional. See Section 8.3.5, “Column Indexes”.

2. BLOB and TEXT columns cannot have DEFAULT values.

**Comment by czs007** [comment]
After discussion. we agreed to use C.int8 as bytes in go for now.
In the later we will support DATATYPE_Uint8.

At that time, we will change to uint8 at the bottom of the storage.
And both int8 and uint8 can be converted to byte.

**Pull request merged**